backend/controller/evaljob_controller.go

```diff
@@ -0,0 +1,568 @@
+	var runAsUser int64 = 1001030000
+
+	// Then compose the Pod CR
+	pod := corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      job.Name,
+			Namespace: job.Namespace,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: job.APIVersion,
+					Kind:       job.Kind,
+					Name:       job.Name,
+					Controller: &ownerRefController,
+					UID:        job.UID,
+				},
+			},
+			Labels: map[string]string{
+				"app.kubernetes.io/name": "fms-lm-eval-service",
+			},
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:            "driver",
+					Image:           r.options.DriverImage,
+					ImagePullPolicy: corev1.PullAlways,
+					Command:         []string{DriverPath, "--copy", DestDriverPath},
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						RunAsUser:                &runAsUser,
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "shared",
+							MountPath: "/opt/app-root/src/bin",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:            "main",
+					Image:           r.options.PodImage,
+					ImagePullPolicy: corev1.PullAlways,
+					Env: []corev1.EnvVar{
+						{
+							Name: "GENAI_KEY",
+							ValueFrom: &corev1.EnvVarSource{
+								SecretKeyRef: &corev1.SecretKeySelector{
+									Key: "key",
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "genai-key",
+									},
+								},
+							},
+						},
+					},
+					Command: generateCmd(job),
+					Args:    generateArgs(job),
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						RunAsUser:                &runAsUser,
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "shared",
+							MountPath: "/opt/app-root/src/bin",
+						},
+					},
+				},
+			},
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: &runAsNonRootUser,
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: corev1.SeccompProfileTypeRuntimeDefault,
+				},
+			},
+			ServiceAccountName: r.options.DriverServiceAccount,
+			Volumes: []corev1.Volume{
+				{
+					Name: "shared", VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+	return &pod
+}
+
+func generateArgs(job *lmevalservicev1beta1.EvalJob) []string {
+	if job == nil {
+		return nil
+	}
+
+	cmds := make([]string, 0, 10)
+	cmds = append(cmds, "python", "-m", "lm_eval", "--output_path", "/opt/app-root/src/output")
```

Comment:
I think this will need to be a mounted volume and/or configurable. A user will need to be able to get the output results stored somewhere.

Reply:
yes, let me create an issue for this feature too
#9